packages/react-native-web/src/exports/Image/types.js

```diff
@@ -102,17 +103,26 @@ export type ImageStyle = {
   tintColor?: ColorValue
 };
 
+export type SourceState = {
+  status: 'IDLE' | 'LOADING' | 'LOADED' | 'ERRORED',
+  source: ImageSource
+};
+
+export type ImageLoadingProps = {|
+  onLoad?: (e: { nativeEvent: ImageResult }) => void,
+  onLoadStart?: () => void,
+  onLoadEnd?: () => void,
+  onError?: ({ nativeEvent: { error: string } }) => void,
+  onProgress?: (e: any) => void
+|};
+
 export type ImageProps = {
-  ...ViewProps,
+  ...$Exact<ViewProps>,
+  ...ImageLoadingProps,
```

Comment:
This is again Flow type related 

In order to extended (spread) more than one types they have to be exact
(exact means the type has only these properties, while inexact would mean it has at least these properties)